portfolio/src/main/webapp/script.js

```diff
@@ -42,18 +49,24 @@ try {
 
 // Hides dropdown menu and shows main view elements.
 const hideMenuShowMain = () => {
-  dropdownMenu.style.display = 'none';
+  ensureHasClass(dropdownMenu, 'show');
+  dropdownMenu.classList.remove('show');
+  dropdownMenu.classList.add('hide');
+
   for (let mainView of mainViews) {
     mainView.style.display = 'flex';
   }
 }
 
 // Hides main view elements and shows dropdown menu.
 const hideMainShowMenu = () => {
```

Comment:
Instead of having two functions for this, can we have one function, e.g, `toggleMainView()`, or `showMainView(shouldShow)`.

Reply:
Sounds good! :D

Reply:
Wait... Sorry but I just realised that this wouldn't make the code any more succinct. Furthermore, I feel that closeButton and menuIcon has only one purpose: to close/open the menu. In this case, why is toggle a good idea? Thanks in advance!

Reply:
I think the issue right now is that I'm seeing quite some repetitive code, for example, instead of doing 
```
dropdownMenu.classList.remove('hide');
dropdownMenu.classList.add('show');
``` 
You can use `dropdownMenu.classList.replace('hide', 'show')`. Also, the replace api is safe enough to no-op if the class to be replaced doesn't exist. https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/replace

Another thing is that the following lines should not have been in global scope in the first place, and they should be only declared inside the functions as they're not necessarily to be a global variable. That would indeed save you some code if you have only one function for it.
```
const mainViews = document.getElementsByClassName('main-view');
const dropdownMenu = document.getElementById('dropdown-menu');
try {
  ensureNonNull(mainViews, dropdownMenu);
} catch (err) {
  console.error(err);
}
```

About toggle, yeah that's exactly why i used `toggleMainView`, instead of `toggleCloseButton`. But if toggle isn't a good name for you, showMainView(shouldShow) is definitely a good way to go.

Reply:
Okay, thanks for the clarification! (: